easyfinance.client/src/app/features/income/list-incomes/list-incomes.component.ts

```diff
@@ -162,4 +180,34 @@ export class ListIncomesComponent implements OnInit {
   previous() {
     this.router.navigate(['/projects', this.projectId]);
   }
+
+
+  getFormFieldErrors(fieldName: string): string[] {
+    const control = this.incomeForm.get(fieldName);
+    const errors: string[] = [];
+
+    if (control && control.errors) {
+      for (const key in control.errors) {
+        if (control.errors.hasOwnProperty(key)) {
+          switch (key) {
+            case 'required':
+              errors.push('This field is required.');
+              break;
+            case 'pattern':
+              if (fieldName === 'amount') {
+                errors.push('Invalid amount format. (0000,00)');
+              }
+              if (fieldName === 'date') {
+                errors.push('Invalid date format.');
+              }
```

Comment:
```suggestion
```
I suggest remove this, because after migrate to material we don't need it anymore.